### PR TITLE
[modules] wing-rotation controller with servo

### PR DESCRIPTION
**Pull request by dewagter**

### CODE REVIEW

**Comment by dewagter** [comment]
How do you enable ```ADC_5``` in the ```test``` architecture?

**Comment by fvantienen** [comment]
You should make `ADC_CHANNEL_WING_ROTATION_POSITION` a configure. Then you can do something similar to:
```
    <configure name="ADC_CHANNEL_WING_ROTATION_POSITION" default="ADC_5" case="lower|upper"/>
    <define name="ADC_CHANNEL_WING_ROTATION_POSITION" value="$(ADC_CHANNEL_WING_ROTATION_POSITION_UPPER)"/>
    <define name="USE_$(ADC_CHANNEL_WING_ROTATION_POSITION_UPPER)"/>
```

**Comment by fvantienen** [comment]
That way the ADC is enabled by itself instead of doing in the airframe

**Comment by fvantienen** [comment]
Btw the naming of these defines etc. could be a lot better. Usually it is best to start with the module name, such that they are all grouped together

File: sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.c [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+void wing_rotation_init(void)
+{
+  // your init code here
+#if !USE_NPS
+  adc_buf_channel(ADC_CHANNEL_WING_ROTATION_POSITION, &buf_wing_rot_pos, ADC_CHANNEL_WING_ROTATION_POSITION_NB_SAMPLES);
+#endif
+
+  // Init wing_rotation_controller struct
+  wing_rotation.servo_pprz_cmd = 0;
```
Comment (fvantienen) [inline sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.c:95]: Just do `wing_rotation = {0};` after that just set the variables which are not `0` or `false`

File: sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.c [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+  wing_rotation.adc_wing_rotation = 0;
+  wing_rotation.wing_angle_rad = 0;
```
Comment (fvantienen) [inline sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.c:97]: Is it really needed to have everything both in rad and deg?

File: sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+#include "std.h"
+
+extern void wing_rotation_init(void);
+extern void wing_rotation_periodic(void);
+extern void wing_rotation_event(void);
+extern bool set_wing_rotation_scheduler(bool rotation_scheduler_on);
+extern bool set_wing_rotation_scheduler_nav(bool rotation_scheduler_on);
+
+// Paramaters
+struct wing_rotation_controller {
```
Comment (fvantienen) [inline sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.h:38]: This struct seems to contain way too many variables which are not needed.

**Review by fvantienen — CHANGES_REQUESTED** [review]

File: sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.h (hunk 3 above)
Comment (dewagter) [reply sw/airborne/modules/rot_wing_drone/wing_rotation_controller_servo.h:38]: removed unneeded: all the rest is used.

**Comment by dewagter** [comment]
@fvantienen update complete

**Comment by gautierhattenberger** [comment]
is is still a draft ?

**Comment by dewagter** [comment]
Working on it. Want to convert it into an actuator_soft_servo if you agree.

**Comment by dewagter** [comment]
Cleanup complete.

**Review by gautierhattenberger — APPROVED** [review]

**Pull request merged**